modules/miner/miner.go

```diff
@@ -194,18 +210,21 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
+		return err
+	}
+
 	m.mu.Lock()
 	defer m.mu.Unlock()
```

Comment:
Only a read lock is necessary here, and only while calling `saveSync`. I don't think it matters much though.